applications/OptimizationApplication/custom_utilities/response/max_overhang_response_utils.h

```diff
@@ -0,0 +1,84 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Reza Najian Asl
+//
+
+#pragma once
+
+// System includes
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+// Application includes
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+class KRATOS_API(OPTIMIZATION_APPLICATION) MaxOverhangAngleResponseUtils
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    using IndexType = std::size_t;
+
+    using GeometryType = ModelPart::ElementType::GeometryType;
+
+    typedef array_1d<double, 3> array_3d;
+
+    using SensitivityFieldVariableTypes = std::variant<const Variable<double>*, const Variable<array_1d<double, 3>>*>;
+
+    using SensitivityVariableModelPartsListMap = std::unordered_map<SensitivityFieldVariableTypes, std::vector<ModelPart*>>;
+
+    ///@}
+    ///@name Static operations
+    ///@{
+
+    static double CalculateValue(const std::vector<ModelPart*>& rModelParts, const Parameters& rResponseSettings);
+
+    static void CalculateSensitivity(
+        const std::vector<ModelPart*>& rEvaluatedModelParts,
+        const SensitivityVariableModelPartsListMap& rSensitivityVariableModelPartInfo,
+        const Parameters& rResponseSettings);
+
+    ///@}
+private:
+    ///@name Private static operations
+    ///@{
+
+    static double CalculateConditionValue(Condition& rCondition, const Parameters& rResponseSettings);
```

Comment:
```suggestion
    static double CalculateConditionValue(
            const Condition& rCondition, 
            const Parameters& rResponseSettings);
```

Style guide and const